packages/core/src/types/IImageVolume.ts

```diff
@@ -1,6 +1,17 @@
 import type { vtkImageData } from '@kitware/vtk.js/Common/DataModel/ImageData';
 import { Metadata, Point3, IImageLoadObject, Mat3 } from '../types';
 
+type VolumeTimePoints = {
+  timePoints: {
+    /** imageIds of each timepoint  */
+    imageIds: Array<string>;
+    /** volume scalar data  */
+    scalarData: Float32Array | Uint8Array;
+  }[];
```

Comment:
so the volume has a root `imageIds` right now, you are saying we add another one `timePoints.imageIds`? What do you think about using the same `imageIds` and `scalarData` and make them [] for 3D and [][] for 4D?

Reply:
Removed! `timePoints` is now private to 4D volumes only